### PR TITLE
A single lyric endpoint(get)

### PR DESCRIPTION
**Pull request by Efoblack**

### CODE REVIEW

**Comment by kali-physi-hacker** [comment]
> _No description provided._

Always please give a description.

File: api/serializers.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from rest-framework import serializers
+
+from .models import Lyric
+
+class LyricSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Lyric
+        fields = ['id', 'title', 'discription', 'content', 'author', 'composer', 'dated_created']
```
Comment (kali-physi-hacker) [inline api/serializers.py:8]: `LyricSerializer` has been implemented by @RJ-0605 already.
Also, you didn't have to explicitly add `id`, we don't need to accept `date_created` also so you didn't need to add that too. Also, check your spellings. Thanks.
Remove the `serializers.py`

File: api/views.py [hunk 2]
```diff
@@ -1,7 +1,25 @@
...
 
 
 def test(request):
     template = ""
     context = {}
     return render(request=request, template_name=template, context=context)
+    from django.shortcuts import render
+
+
+@csrf_exempt
```
Comment (kali-physi-hacker) [inline api/views.py:16]: we didn't need this `decorator`

File: api/views.py [hunk 3]
```diff
@@ -1,7 +1,25 @@
...
+def lyric_detail(request, pk):
+    try:
+        lyric = Lyric.objects.get(pk=pk)
+    except Lyric.DoesNotExist:
+        return HttpResponse(status=404)
+
+    if request.method == "GET":
+        serializer = LyricSerializer(lyric)
```
Comment (kali-physi-hacker) [inline api/views.py:24]: Also, the `api_view` decorator???

**Review by kali-physi-hacker — CHANGES_REQUESTED** [review]

File: api/views.py [hunk 4]
```diff
@@ -1,7 +1,23 @@
...
     context = {}
     return render(request=request, template_name=template, context=context)
+
+
+@api_view(["GET"])
+def lyric_detail(request, pk):
+    try:
+        lyric = Lyric.objects.get(pk=pk)
+    except Lyric.DoesNotExist:
+        return HttpResponse(status=404)
```
Comment (kali-physi-hacker) [inline api/views.py:19]: that line of code should not exist. We're writing apis, as such any response should be a JsonResponse. So basically, that line should be `Response(data={"error": "Lyric not found"}, status=status.HTTP_404_NOT_FOUND)`

File: api/views.py [hunk 5]
```diff
@@ -1,7 +1,23 @@
...
+
+    if request.method == "GET":
+        serializer = LyricSerializer(lyric)
+        return JsonResponse(serializer.data)
```
Comment (kali-physi-hacker) [inline api/views.py:23]: hmmm, You're supposed to leverage `djangorestframework's` `Response` class. Also, I remember also saying that on rule of thumb or **code best practice** to follow is: **explicit is better than implicit**, so in all, that line of code should be: `Response(data=serializer.data, status=status.HTTP_200_OK)

**Review by kali-physi-hacker — CHANGES_REQUESTED** [review]

File: api/views.py (hunk 5 above)
Comment (kali-physi-hacker) [reply api/views.py:23]: **Explicit is better than implicit**: https://www.python.org/dev/peps/pep-0020/. After you're done, check that out. It's called the **Zen of python**.

File: api/urls.py [hunk 6]
```diff
@@ -1,5 +1,9 @@
 from django.urls import path
+from .views import lyric_detail
 
 from .views import test
 
-urlpatterns = [path("/test/", test, name="test")]
+urlpatterns = [
+    path("/test/", test, name="test"),
+    path("/api/lyric/<int:id>/", views.lyric_detail),
```
Comment (kali-physi-hacker) [inline api/urls.py:8]: you should change this to `/lyric/<int:id>/`

File: api/views.py [hunk 7]
```diff
@@ -1,7 +1,30 @@
...
+
+    if request.method == "GET":
+        serializer = LyricSerializer(lyric)
+        return Response(data=serializer.data, status=status.HTTP_200_OK)
+
+    elif request.method == "PUT":
+        serializer = LyricSerializer(lyric, data=request.data)
+        if serializer.is_valid():
+            return Response(serializer.data)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```
Comment (kali-physi-hacker) [inline api/views.py:30]: explicit, stuff here????

**Review by kali-physi-hacker — CHANGES_REQUESTED** [review]

File: api/views.py (hunk 4 above)
Comment (kali-physi-hacker) [reply api/views.py]: is this resolved?

File: api/views.py (hunk 5 above)
Comment (kali-physi-hacker) [reply api/views.py]: is this done?

File: api/urls.py (hunk 6 above)
Comment (kali-physi-hacker) [reply api/urls.py]: resolved?

File: api/views.py (hunk 7 above)
Comment (kali-physi-hacker) [reply api/views.py:32]: resolved?

Comment (Efoblack) [reply api/views.py]: Made the changes now

File: api/urls.py (hunk 6 above)
Comment (Efoblack) [reply api/urls.py]: Yeah

File: api/views.py (hunk 4 above)
Comment (Efoblack) [reply api/views.py]: yeah

File: api/views.py (hunk 5 above)
Comment (Efoblack) [reply api/views.py]: yeah

File: api/views.py [hunk 8]
```diff
@@ -1,7 +1,32 @@
...
```
Comment (kali-physi-hacker) [inline api/views.py:32]: you're not listening, explicit here?????????

**Pull request merged**